src/hw_isolation_record/manager.cpp

```diff
@@ -532,12 +531,43 @@ void Manager::restore()
     });
 }
 
+void Manager::processHardwareIsolationRecordFile()
+{
+    /**
+     * Start timer in the event loop to get the final isolated hardware
+     * record list which are updated by the host because of the atomicity
+     * on the partition file (which is used to store isolated hardware details)
+     * between BMC and Host.
+     */
+    try
+    {
+        _timerObjs.emplace(
+            std::make_unique<
+                sdeventplus::utility::Timer<sdeventplus::ClockId::Monotonic>>(
+                _eventLoop,
+                std::bind(std::mem_fn(&hw_isolation::record::Manager::
+                                          handleHostIsolatedHardwares),
+                          this),
+                std::chrono::seconds(5)));
+    }
+    catch (const std::exception& e)
+    {
+        log<level::ERR>(
+            fmt::format("Exception [{}], Failed to process "
+                        "hardware isolation record file that's updated",
+                        e.what())
+                .c_str());
+    }
+}
+
 void Manager::handleHostIsolatedHardwares()
 {
-    // Wait for some time to get the final isolated hardware record list
-    // which are updated by the host because of the atomicity on partition
-    // file (which is used to isolated hardware details) between BMC and Host.
-    sleep(5);
+    auto timerObj = std::move(_timerObjs.front());
+    _timerObjs.pop();
+    if (timerObj->isEnabled())
+    {
+        timerObj->setEnabled(false);
+    }
```

Comment:
do we need this any how the timerobj once it goes out of scope will be deleted? and will not fire any more

Reply:
if it takes more than 5 sec to exit then, the timer will call again but, we don't need to do that since we are already in the mid of the process.
FYR: https://www.freedesktop.org/software/systemd/man/sd_event_source_set_enabled.html

By default, [sdeventplus setting is enabled](https://github.com/openbmc/sdeventplus/blob/master/src/sdeventplus/utility/timer.cpp#L31) since we are passing the interval while creating the timer object.